src/lib.rs

```diff
@@ -260,13 +293,8 @@ pub fn eval_beat<T: Into<Val>>(program: &Program, t: T) -> Result<Val, ()> {
                 })
             }
             Arr(size) => {
-                let index: i64 = stack.pop().ok_or(())?.into();
-                // We need to pop `size` values, so our stack needs to be
-                // atleast size elements long. Note that split_off panics if we
-                // exceed the length of the vector, so we need this if guard.
-                if size > stack.len() {
-                    return Err(());
-                } else if size == 0 {
+                let index: i64 = stack.pop().unwrap().into();
+                if size == 0 {
```

Comment:
Since we don't need validation anymore, we could handle this with another pattern above this one, `Arr(0) => { … }` before `Arr(size) => { … }`.